tasks/build.py

```diff
@@ -407,6 +422,17 @@ def download_pr(repo_name, branch_name, pr, arch_job_dir):
         error_stage = _ERROR_CURL
         return curl_output, curl_error, curl_exit_code, error_stage
 
+    git_diff_cmd = ' '.join([
+        f"git fetch origin pull/{pr.number}/head:{pr.number}",
+        f"git diff HEAD pr{pr.number} > {pr.number}.diff",
```

Comment:
if you're now creating the like this, should'nt you remove the `curl_cmd` above?

Reply:
Yes, of course, I'll tackle that too

Reply:
also, this is not exactly the same as the downloading the patch and applying it, as there can be changes pushed to the repo *after* the PR was created, which will be undone by this?

Reply:
@smoors made a good point (in Slack) about this change: this doesn't take into account that there may be changes in the `main` branch that are not included in the PR branch yet.

So, a better approach would be to:
* fetch the PR branch
* check out the PR branch
* `git merge main`
* swap back to `main` branch
* `git diff`
* `git apply`

Reply:
This procedure hasn't been implemented yet, right?

A few thoughts:
- `main` should probably be the target branch of the PR, for example, for EESSI/2023.06 that would be 2023.06-software.eessi.io?
- the `git diff` after the swap back to "main" is really just the plain `git diff` or rather `git diff PR_branch`?

Reply:
I'll change this so the new approach based on `git` is only used when `clone_via` is set to `'ssh'` (so nothing changes for existing bots that listen to a public repo).

That gives us some freedom to experiment with the situation where the target branch has been updated since the branch for the PR was created.

Reply:
I've updated this in 80efece to only use the `git diff` approach when `clone_git_repo_via` is used.

It's still doing only `git fetch` + `git diff`, so you could still have issues when the PR branch is way behind on the target branch, I'll try to look into that next.